aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -557,5 +557,39 @@ Tensor& NestedTensor_mul__Tensor(Tensor& self, const Tensor& other) {
       });
 }
 
+Tensor select_nested(const Tensor& self, int64_t dim, int64_t index) {
+  TORCH_CHECK(
+    dim == 0,
+    "NestedTensor can only be selected along dimension 0 ",
+    "got dimension ", dim, " instead."
+  );
+  // size_matrix.dim() can only have 2 possible values:
+  //     * 2, as expected
+  //     * 0, when the nested tensor is empty
+  Tensor sizemat = get_nested_size_tensor(self);
+  TORCH_CHECK(
+    sizemat.dim() == 2,
+    "cannot index an empty nested tensor."
+  );
+  // nested_tensor[i] = i-th original tensor
+  int64_t ntensors = sizemat.size(0);
```

Comment:
Until we add support for [`size.int`](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L4284-L4288) to NestedTensor itself (note that this is different from `sizes`) you can also use [`opt_size`](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/NestedTensorImpl.h#L25-L31) to get this information. 

As an aside, a convenient way of defining `size.int` is as
```
int64_t size(const Tensor& self, int64_t dim) {
    return *get_nested_tensor_impl(self)->opt_sizes(dim)
}
```

This will throw an error message if `size` has been requested for an irregular dimension. It'll also allow a user to construct `sizes` *if all dimensions are regular* at which point a NestedTensor duck types to become a regular Tensor.